shell/gpu/gpu_surface_metal.mm

```diff
@@ -89,30 +82,17 @@
       return false;
     }
 
-    const auto has_external_view_embedder = delegate_->GetExternalViewEmbedder() != nullptr;
-
     auto command_buffer =
         fml::scoped_nsprotocol<id<MTLCommandBuffer>>([[command_queue_.get() commandBuffer] retain]);
 
     fml::scoped_nsprotocol<id<CAMetalDrawable>> drawable(
         reinterpret_cast<id<CAMetalDrawable>>(next_drawable_));
     next_drawable_ = nullptr;
 
-    // External views need to present with transaction. When presenting with
-    // transaction, we have to block, otherwise we risk presenting the drawable
-    // after the CATransaction has completed.
-    // See:
-    // https://developer.apple.com/documentation/quartzcore/cametallayer/1478157-presentswithtransaction
-    // TODO(dnfield): only do this if transactions are actually being used.
-    // https://github.com/flutter/flutter/issues/24133
-    if (!has_external_view_embedder) {
-      [command_buffer.get() presentDrawable:drawable.get()];
-      [command_buffer.get() commit];
-    } else {
-      [command_buffer.get() commit];
-      [command_buffer.get() waitUntilScheduled];
-      [drawable.get() present];
-    }
+    [command_buffer.get() commit];
+    [command_buffer.get() waitUntilScheduled];
+    [drawable.get() present];
```

Comment:
I'm pretty sure this will break platform views - have you tested this with a platform view?  e.g. run the webview example on a local engine built with metal support.

Reply:
If anything, this brings the behavior more in line with the condition where there are platform views. When there is a platform view, `has_external_view_embedder` in the previous code path will be `true`. Now, both paths do the same thing where they wait for the GPU commands to be flushed in the current transaction.

Reply:
And yes. I did test this locally with platform views enabled (specifically the web view example).

Reply:
Ahhh duh I misread the old code.